commit f55dd8ef5baddbda3a7c5332ec1c2cbc4c57e400
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Feb 20 10:07:27 2018 +0000

    i2c-gpio: Also set bus numbers from reg property
    
    I2C busses can be assigned specific bus numbers using aliases in
    Device Tree - string properties where the name is the alias and the
    value is the path to the node. The current DT parameter mechanism
    does not allow property names to be derived from a parameter value
    in any way, so it isn't possible to generate unique or matching
    aliases for nodes from an overlay that can generate multiple
    instances, e.g. i2c-gpio.
    
    Work around this limitation (at least temporarily) by allowing
    the i2c adapter number to be initialised from the "reg" property
    if present.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/i2c/busses/i2c-gpio.c b/drivers/i2c/busses/i2c-gpio.c
index 58abb3eced58..8da45f708ca6 100644
--- a/drivers/i2c/busses/i2c-gpio.c
+++ b/drivers/i2c/busses/i2c-gpio.c
@@ -321,7 +321,9 @@ static int i2c_gpio_probe(struct platform_device *pdev)
 	adap->dev.parent = dev;
 	adap->dev.of_node = np;
 
-	adap->nr = pdev->id;
+	if (pdev->id != PLATFORM_DEVID_NONE || !pdev->dev.of_node ||
+	    of_property_read_u32(pdev->dev.of_node, "reg", &adap->nr))
+		adap->nr = pdev->id;
 	ret = i2c_bit_add_numbered_bus(adap);
 	if (ret)
 		return ret;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1141_f55dd8ef5baddbda3a7c5332ec1c2cbc4c57e400.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1141_f55dd8ef5baddbda3a7c5332ec1c2cbc4c57e400.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
